app/components/network_indicator/network_indicator.js

```diff
@@ -0,0 +1,390 @@
+        }).isRequired,
+        isConnecting: PropTypes.bool,
+        isLandscape: PropTypes.bool,
+        isOnline: PropTypes.bool,
+        webSocketStatus: PropTypes.string,
+    };
+
+    static defaultProps = {
+        isOnline: true,
+    };
+
+    static contextTypes = {
+        intl: intlShape.isRequired,
+    };
+
+    static getDerivedStateFromProps(nextProps, prevState) {
+        const {isLandscape, networkStatus} = prevState;
+
+        if (nextProps.isLandscape !== isLandscape && networkStatus) {
+            const navBar = this.getNavBarHeight(nextProps.isLandscape);
+            const top = new Animated.Value(navBar - HEIGHT);
+
+            return {
+                navBar,
+                top,
+            };
+        }
+
+        return null;
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.isX = DeviceInfo.getModel().includes('iPhone X');
+        const navBar = this.getNavBarHeight(props.isLandscape);
+
+        this.state = {
+            isLandscape: props.isLandscape,
+            networkStatus: null,
+            navBar,
+            top: new Animated.Value(navBar - HEIGHT),
+        };
+
+        this.backgroundColor = new Animated.Value(0);
+
+        this.networkListener = networkConnectionListener(this.handleConnectionChange);
+    }
+
+    componentDidMount() {
+        this.mounted = true;
+    }
+
+    componentDidUpdate(prevProps) {
+        const {webSocketStatus} = this.props;
+
+        if (this.props.isOnline) {
+            if (this.state.networkStatus && webSocketStatus === RequestStatus.STARTED && this.props.webSocketStatus === RequestStatus.SUCCESS) {
+                // Show the connected animation only if we had a previous network status
+                this.connected();
+            } else if (webSocketStatus === RequestStatus.STARTED && this.props.webSocketStatus === RequestStatus.FAILURE && this.props.isConnecting) {
+                // Show the connecting bar if it failed to connect at least twice
+                this.connecting();
+            }
+        } else if (prevProps.isOnline && !this.props.isOnline) {
+            this.offline();
+        }
+    }
+
+    componentWillUnmount() {
+        this.mounted = false;
+
+        const {closeWebSocket, stopPeriodicStatusUpdates} = this.props.actions;
+
+        this.networkListener.removeEventListener();
+
+        AppState.removeEventListener('change', this.handleAppStateChange);
+
+        closeWebSocket();
+        stopPeriodicStatusUpdates();
+    }
+
+    connect = () => {
+        // need to change this to use the non-listener function
+        checkConnection((result) => {
+            this.setLocalState({networkStatus: CONNECTING}, () => {
+                if (result) {
+                    this.props.actions.connection(true);
+                    this.initializeWebSocket();
+                } else {
+                    this.setLocalState({networkStatus: OFFLINE});
```

Comment:
shouldn't we here set the connection as false as well as to close the WS?

Reply:
Yes, fixed.